### DISCUSSION TITLE
How to upload images using activitypub?

### DISCUSSION THREAD

**Discussion by springrain** [mastodon/mastodon#25040] [Q&A] [answered]
I am using Golang to develop some APIs for mastodon. How can I upload images to mastodon using Activitypub? I couldn't find any relevant content in the document. Thank you very much for your help

**Comment by trwnh** [comment, marked as answer]
mastodon does not support the activitypub client-to-server protocol. the expectation is to use the Mastodon REST API instead.

https://docs.joinmastodon.org/methods/media/

https://docs.joinmastodon.org/methods/statuses/#create (see `media_ids[]` parameter)
